### PR TITLE
exporter: framework for componentizing TabletServer

### PR DESCRIPTION
**Pull request by sougou**
The new framework allows you to create a named environment under which all stats variables and http endpoints get remapped to distinct URLs or var dimensions.

Details are in #5791

### CODE REVIEW

File: go/vt/servenv/status.go [hunk 1]
```diff
@@ -1,13 +1,13 @@
 /*
-Copyright 2019 The Vitess Authors.
+Copyright 2017 Google Inc.
```
Comment (morgo) [inline go/vt/servenv/status.go:2]: Looks like an error here :-)

Comment (sougou) [reply go/vt/servenv/status.go:2]: Indeed :)

File: go/vt/servenv/exporter.go [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+// because we build derived variables like Rates, which need to continue
+// referencing the original variable that was created.
+type Exporter struct {
+	name, label string
+	handleFuncs map[string]*handleFunc
+	sp          *statusPage
+}
+
+// NewExporter creates a new Exporter with name as namespace.
+// label is the name of the additonial dimension for the stats vars.
```
Comment (harshit-gangal) [inline go/vt/servenv/exporter.go:103]: typo "additonial" -> "additional"

**Comment by harshit-gangal** [comment]
Overall looks good to me. I have to go over the RFC to get better clarity.

File: go/vt/servenv/exporter.go [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+	// exportedTimingsVars contains all the timings vars.
+	exportedTimingsVars = make(map[string]*stats.MultiTimings)
+
+	// exportedOtherStatsVars contains Rates, Histograms and Publish vars.
+	exportedOtherStatsVars = make(map[string]*expvar.Map)
+)
+
+//-----------------------------------------------------------------
+
+// Exporter remaps http and stats end-points to distinct namespaces.
```
Comment (deepthi) [inline go/vt/servenv/exporter.go:68]: Nice docs.

**Pull request merged**